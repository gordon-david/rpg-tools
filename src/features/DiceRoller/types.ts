export interface Dice {
    id: string
}