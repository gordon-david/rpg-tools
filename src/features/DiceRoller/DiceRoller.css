.dice-roller {
    max-width: 320px;
    margin: 1rem;
    border: 1px solid gold;
}

.dice-roller__dice-grid {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

.dice {
    width: 5rem;
    display: flex;
    flex-direction: column;
}

.dice__amount {
    text-align: center;
}

.dice-roller__controls {
    display: flex;
    justify-content: space-around;
}
