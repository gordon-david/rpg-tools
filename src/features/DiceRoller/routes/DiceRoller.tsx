import { useEffect, useState } from "react";
import { Dice } from "../components/Dice";
import { DicePool } from "../components/DicePool";
import { DiceKey, PolyhedralDicePoolFactory } from "../core/Dice";


export function DiceRoller() {
    const [dicePool, setDicePool] = useState(PolyhedralDicePoolFactory())
    const [results, setResults] = useState([])
    const [diceMap, setDiceMap] = useState({})

    useEffect(() => {
        setDiceMap(dicePool.getDiceMap)
    }, [dicePool])

    function addDice(key: DiceKey){
        // add to dicePool
        dicePool.add(key)
        setDicePool(dicePool)
    }

    function removeDice(key: DiceKey){
        dicePool.remove(key)
        setDicePool(dicePool)
    }

    return <>

        <div className="dice-roller">
            <DicePool />
            <div>Roll Results List</div>
        </div>
    </>
}

