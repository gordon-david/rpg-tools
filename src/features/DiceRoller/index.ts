import { DiceRoller } from "./routes/DiceRoller";
export * from "./types"

export {DiceRoller}