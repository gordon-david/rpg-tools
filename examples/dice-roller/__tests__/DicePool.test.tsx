import React from 'react'
import { shallow, mount } from 'enzyme'
import { DicePool, IDicePoolProps } from '../components/DiceRoller/DicePool'

function makeTestProps(){
    return {
        roll: function(){},
        dice: [{diceType: 'test-dice', displayName:'test-displayname', amount:0}]
    }
}

test('renders correctly', () => {
    const props: IDicePoolProps = makeTestProps()
    const wrapper = shallow(<DicePool roll={props.roll} dice={props.dice} />)
    expect(wrapper).toMatchSnapshot()
})

test('renders with default props', () => {
    const wrapper = shallow(<DicePool />)
    expect(wrapper).toMatchSnapshot()
})

test('constructs dice pool of Dice components from diceTypes props', () => {
    const props = makeTestProps()
    const wrapper = shallow(<DicePool dice={props.dice} roll={props.roll} />)
    expect(wrapper.find(`.dice#${props.dice[0].diceType}`))
})

test('clear button resets dice pool counts to 0', () => {
    const props = makeTestProps()
    const wrapper = mount(<DicePool dice={props.dice} roll={props.roll} />)
    const diceChild = wrapper.find(`.dice#${props.dice[0].diceType}`)

    diceChild.find('button.increment').simulate('click')
    expect(diceChild.find('.amount').text()).toBe('1')
    wrapper.find('button#clear').simulate('click')
    expect(diceChild.find('.amount').text()).toBe('0')
})

test('increments dice pool when child dice increment prop is called', () => {
    const props = makeTestProps()
    const wrapper = mount(<DicePool dice={props.dice} roll={props.roll} />)
    const diceChild = wrapper.find(`.dice#${props.dice[0].diceType}`)

    diceChild.find('button.increment').simulate('click')
    expect(diceChild.find('.amount').text()).toBe('1')
})

// test('roll prop is called when roll button is clicked', () => {
//     const mockFn = jest.fn()
//     const wrapper = shallow(<DicePool roll={mockFn} />)
//     wrapper.find('button#roll').simulate('click')
//     expect(mockFn).toHaveBeenCalledTimes(1)
// })

// test('roll button clears current dice pool')
