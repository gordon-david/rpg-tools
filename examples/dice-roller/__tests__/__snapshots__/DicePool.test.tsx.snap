// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders with default props 1`] = `ShallowWrapper {}`;
